### ISSUE TITLE
is corcel also suited to aid in populating a wordpress database from a laravel backend?

### ISSUE THREAD

**Issue by vesper8** [corcel/corcel#230]
I just found out about Corcel. My plan is to use it to populate content via laravel using it. I think this will work. But am wondering if this is the best tool to do this with since this tool seems to exist to do the exact opposite.

I know I could simply use something like https://github.com/Cyber-Duck/laravel-wp-api or use the wordpress rest api directly. But Corcel seems to be under active development and as some useful features.

Basically I need a tool that will allow me to create/update/delete from all tables including of course posts but also the meta tables, terms, term relationships, and very importantly the options table

Corcel can do all that? Any reason why I wouldn't use it for my needs?

**Comment by jgrossi** [comment]
Hi @vesper8 Corcel is better for retrieving content from WP. It allows you to add posts and custom meta but I cannot say it would work perfectly. There're just a few tests for saving data, but you can give it a try. Maybe if you find some bug you can contribute with it ;-) JG

**Comment by vesper8** [comment]
Thanks! I'm giving it a go, so far it's proven to be quite useful.

I was having a hell of a time creating a new user by passing an array to User::create() because of how you're disabling the updated_at.. it seems to cause a weird bug. I think 

```
    public function getUpdatedAtColumn() {
        return null;
    }
```

was solving the problem that was being caused by

```
const UPDATED_AT = null;

public function setUpdatedAtAttribute($value) {} 
```

in the end just adding 
```
    protected $guarded = [];
```

solved the problems with updated_at as well as the fillable issues

Here's my User model now, I commented the stuff I tried earlier in case it helps someone

```
<?php

namespace App\Models\Wordpress;

use Corcel\User as Corcel;

class User extends Corcel
{
    //const UPDATED_AT = 'updated_at';

    protected $connection = 'wpdb';

    protected $guarded = [];

//    public function getUpdatedAtColumn() {
//        return null;
//    }
}
```

If you'd like I can use this thread to continue to report on any obstacles I get in while populating posts, metas, terms, menus and options so Corcel can be improved to work in both ways

**Comment by vesper8** [comment]
I also wanted to ask in case anyone has ANY insights here. But I was wondering if there is a better way to reconstruct posts and terms and term_relationships. Basically I want to add everything that a popular video-centric wp theme is currently adding via its fancy plugins and custom approach.

Right now I am using dbdiff (awesome!) to show me what rows have been added to all of the wp tables following an event like setting options or creating a custom post. And then I'm using Corcel to programatically do the same inserts to the same tables but using data pulled from my own platform.

Wouldn't happen to know of an even better tool or methodology for doing this? All the options and metas get a little tricky specially considering I haven't produced plugins or worked a lot with these taxonomy and meta tables before.

Thanks in any case!

**Comment by jgrossi** [comment]
@vesper8 Corcel matches the WP database using Laravel Eloquent, so you can use cool stuff the ORM has like this one:

```php
Post::create(...)->meta()->create(..); // both creating `wp_posts` and `wp_postmeta` rows
```

I don't know another tool that do that. I think Corcel is enough for what you want.

JG

**Comment by jgrossi** [comment]
@vesper8 and sure! report any issue here ;-)
